### PR TITLE
Continue build changes and refactoring

### PR DESCRIPTION
**Pull request by adityapatwardhan**

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -24,12 +24,14 @@ Implement build and packaging of the package and place the output $OutDirectory/
...
-    Get-ChildItem -Path $script:ModuleRoot -Filter "$script:ModuleName.ps*1" | ForEach-Object { Copy-Item -Path $_.FullName -Destination $script:OutModule -Verbose }
+    Get-ChildItem -Path $script:ModuleRoot -Filter "*.ps*1" | ForEach-Object { Copy-Item -Path $_.FullName -Destination $script:OutModule -Verbose }
+    Copy-Item -Path (Join-Path $script:ModuleRoot 'yml') -Recurse $script:OutModule -Force
+
     Write-Verbose -Verbose "Ending DoBuild"
 }
 
 #region Special casing for PSPackageProject CI system
-$PSPackageProjectModule = Join-Path $PSScriptRoot $SrcPath -AdditionalChildPath "$ModuleName.psd1"
+$PSPackageProjectModule = [System.IO.Path]::Join($PSScriptRoot, $SrcPath, "$ModuleName.psd1")
```
Comment (adityapatwardhan) [inline build.ps1]:
```suggestion
$PSPackageProjectModule = [System.IO.Path]::Combine($PSScriptRoot, $SrcPath, "$ModuleName.psd1")
```

Comment (rjmholt) [inline build.ps1]: `Path.Join` [isn't available in .NET Framework](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.join?view=netstandard-2.1&viewFallbackFrom=netframework-4.8#System_IO_Path_Join_System_String_System_String_System_String_)

File: src/PSPackageProject.psm1 [hunk 2]
```diff
@@ -545,6 +541,15 @@ Describe "Test ${moduleName}" {
...
     Initialize-CIYml -Path ${moduleRoot}
+
+    # make pspackageproject.json
+    @{
+        SourcePath = 'src'
+        ModuleName = "${ModuleName}"
+        TestPath = 'test'
+        HelpPath = 'help'
+        BuildOutputPath = 'out'
+    } | ConvertTo-Json | Out-File (Join-Path ${moduleRoot} "pspackageproject.json")
```
Comment (rjmholt) [inline src/PSPackageProject.psm1]: Should specify an encoding for Win PS

File: src/PSPackageProject.psm1 [hunk 3]
```diff
@@ -98,11 +98,12 @@ function Initialize-CIYml {
...
-    Copy-Item $boilerplateCIYml -Destination $Path
+    $boilerplateCIYml = Join-Path2 -Path $PSScriptRoot -ChildPath 'yml' -AdditionalChildPath 'ci_for_init.yml'
+    $destYmlPath = New-Item (Join-Path -Path $Path -ChildPath '.ci') -ItemType Directory
+    Copy-Item $boilerplateCIYml -Destination (Join-Path $destYmlPath -ChildPath 'ci.yml') -Force
 
-    $boilerplateTestYml = Join-Path2 -Path $PSScriptRoot -ChildPath 'yml' -AdditionalChildPath 'test.yml'
-    Copy-Item $boilerplateTestYml -Destination $Path
+    $boilerplateTestYml = Join-Path2 -Path $PSScriptRoot -ChildPath 'yml' -AdditionalChildPath 'test_for_init.yml'
+    Copy-Item $boilerplateTestYml -Destination (Join-Path $destYmlPath -ChildPath 'test.yml') -Force
 }
```
Comment (adityapatwardhan) [inline src/PSPackageProject.psm1:107]: Also publish build_for_init.ps1

**Review by rjmholt — APPROVED** [review]

File: build.ps1 [hunk 4]
```diff
@@ -24,12 +24,14 @@ Implement build and packaging of the package and place the output $OutDirectory/
 function DoBuild
 {
     Write-Verbose -Verbose "Starting DoBuild"
-    Get-ChildItem -Path $script:ModuleRoot -Filter "$script:ModuleName.ps*1" | ForEach-Object { Copy-Item -Path $_.FullName -Destination $script:OutModule -Verbose }
+    Get-ChildItem -Path $script:ModuleRoot -Filter "*.ps*1" | ForEach-Object { Copy-Item -Path $_.FullName -Destination $script:OutModule -Verbose }
```
Comment (TravisEz13) [inline build.ps1]:
```suggestion
    Get-ChildItem -Path $script:ModuleRoot -Filter "*.ps*1" | 
        ForEach-Object { Copy-Item -Path $_.FullName -Destination $script:OutModule -Verbose }
```

**Review by TravisEz13 — APPROVED** [review]

File: build.ps1 (hunk 1 above)
Comment (adityapatwardhan) [reply build.ps1]: fixed

File: src/PSPackageProject.psm1 (hunk 2 above)
Comment (adityapatwardhan) [reply src/PSPackageProject.psm1]: Fixed

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**